client/src/connection/ssh/index.ts

```diff
@@ -25,175 +18,199 @@ export interface Config {
-          runResult.html5 = fileContents;
-          runResult.title = "Result";
-        }
-      }
-      resolve?.(runResult);
+      this.setTimer();
+      this.conn.connect(cfg);
     });
-  });
-}
+  };
+  run(code: string, onLog?: (logs: LogLine[]) => void): Promise<RunResult> {
+    this.onLog = onLog;
+    this.html5FileName = "";
+    if (this.logs.length) {
+      this.onLog?.(this.logs.map((line) => ({ type: "normal", line })));
+      this.logs = [];
+    }
 
-function setup(): Promise<void> {
-  return new Promise((pResolve, pReject) => {
-    resolve = pResolve;
-    reject = pReject;
+    return new Promise((_resolve, _reject) => {
+      this.resolve = _resolve;
+      this.reject = _reject;
 
-    if (stream) {
-      resolve();
+      this.stream?.write(`${code}\n`);
+      this.stream?.write(`%put ${endCode};\n`);
+    });
+  }
+  close(): void | Promise<void> {
+    if (!this.stream) {
       return;
     }
-
-    const cfg: ConnectConfig = {
-      host: config.host,
-      port: config.port,
-      username: config.username,
-      privateKey: readFileSync(config.privateKeyPath),
-      readyTimeout: sasLaunchTimeout,
-    };
-    conn.connect(cfg);
-  });
-}
-
-function run(
-  code: string,
-  onLogFn?: (logs: LogLine[]) => void
-): Promise<RunResult> {
-  onLog = onLogFn;
-  html5FileName = "";
-  if (logs.length) {
-    onLog?.(logs.map((line) => ({ type: "normal", line })));
-    logs = [];
+    this.stream.write("endsas;\n");
+    this.stream.end("exit\n");
   }
 
-  return new Promise((_resolve, _reject) => {
-    resolve = _resolve;
-    reject = _reject;
-
-    stream?.write(`${code}\n`);
-    stream?.write(`%put ${endCode};\n`);
-  });
-}
+  setTimer() {
+    this.clearTimer();
+    this.timer = setTimeout(() => {
+      this.reject?.(
+        new Error("Failed to connect to Session. Check profile settings.")
+      );
+      this.timer = undefined;
+      this.close();
+    }, sasLaunchTimeout);
+  }
 
-function close() {
-  if (!stream) {
-    return;
+  clearTimer() {
+    this.timer && clearTimeout(this.timer);
+    this.timer = undefined;
   }
-  stream.write("endsas;\n");
-  stream.end("exit\n");
-}
+  getResult() {
+    const runResult: RunResult = {};
+    if (!this.html5FileName) {
+      this.resolve?.(runResult);
+      return;
+    }
+    let fileContents = "";
+    this.conn.exec(
+      `cat ${this.html5FileName}.htm`,
+      (err: Error, s: ClientChannel) => {
+        if (err) {
+          this.reject?.(err);
+          return;
+        }
 
-export function getSession(c: Config): Session {
-  config = c;
-  return {
-    setup,
-    run,
-    close,
-  };
+        s.on("data", (data) => {
+          fileContents += data.toString().trimEnd();
+        }).on("close", (code) => {
+          const rc: number = code;
+
+          if (rc === 0) {
+            //Make sure that the html has a valid body
+            //TODO: should this be refactored into a shared location?
```

Comment:
I feel like I've asked something like this before, but do we have an issue or know how we'll make a decision on this? If we _have_ to leave a todo in code, I'd really like for it to reference an issue so we don't forget it.